structs.asm

```diff
@@ -187,6 +193,9 @@ MACRO dstruct ; struct_type, instance_name[, ...]
     ; there because that would require a duplicated `ELSE`).
     DEF IS_NAMED_INSTANTIATION = 0
     IF _NARG > 2
+        IF DEF(\1_DISABLE_INITIALIZER)
+            FAIL "Structs containing a union cannot be initialized using dstruct."
```

Comment:
Alternatively, instead of defining this variable, check if a field has a size of 0 (this can be done in the "main" loop), and abort then.